pkg/endpoint/endpoint.go

```diff
@@ -2622,7 +2631,59 @@ func (e *Endpoint) syncPolicyMapController() {
 // This function must be called after re-aquiring the endpoint mutex to verify
 // that the endpoint has not been removed in the meantime.
 //
-// endpoint.Mutex must be held
+// endpoint.mutex must be held in read mode at least
 func (e *Endpoint) IsDisconnecting() bool {
 	return e.state == StateDisconnected || e.state == StateDisconnecting
 }
+
+// LockAlive returns error if endpoint was removed, locks underlying mutex otherwise
+func (e *Endpoint) LockAlive() error {
+	e.mutex.Lock()
+	if e.IsDisconnecting() {
+		e.mutex.Unlock()
+		return fmt.Errorf("endpoint was removed before lock attempt")
```

Comment:
`lock failed: endpoint is in the process of being removed`

Reply:
done